pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -6921,3 +6894,64 @@ func (r *ClickHouseReader) PreloadMetricsMetadata(ctx context.Context) []error {
 
 	return errorList
 }
+
+func (r *ClickHouseReader) GetUpdatedMetricsMetadata(ctx context.Context, metricNames ...string) (map[string]*model.UpdateMetricsMetadata, *model.ApiError) {
+	cachedMetadata := make(map[string]*model.UpdateMetricsMetadata)
+	var missingMetrics []string
+
+	// First, try retrieving each metric from cache.
+	for _, metricName := range metricNames {
+		metadata := new(model.UpdateMetricsMetadata)
+		cacheKey := constants.UpdatedMetricsMetadataCachePrefix + metricName
+		retrieveStatus, err := r.cache.Retrieve(ctx, cacheKey, metadata, true)
+		if err == nil && retrieveStatus == cache.RetrieveStatusHit {
+			cachedMetadata[metricName] = metadata
+		} else {
+			if err != nil {
+				zap.L().Info("Error retrieving metrics metadata from cache", zap.String("metric_name", metricName), zap.Error(err))
+			} else {
+				zap.L().Info("Cache miss for metrics metadata", zap.String("metric_name", metricName))
```

Comment:
This is the spam that was mentioned in the channel. Imagine a metric name that doesn't require an update to metadata. Now each time when a metric is queried, it calls `GetUpdatedMetricsMetadata` which wouldn't have the metric in cache because it was never updated. And it will keep writing useless log each time.